<launch>

    <node pkg="djiros" type="djiros_node" name="djiros" output="screen" launch-prefix="" required="true">
        <param name="log_level" type="string" value="info"/>
        <param name="serial_name" type="string" value="/dev/tty.dji.n3"/>
        <param name="baud_rate" type="int" value="921600"/>
        <param name="app_id"        type="int"      value="1024795"/>
        <param name="app_version"   type="int"      value="1"/>
        <param name="enc_key"       type="string"   value="a5f11a418f3dcdc39c4a1efb35a9dd4545e47b56cae9d98c988468277a9305dc"/>
        <param name="sensor_mode" type="bool" value="false"/>
        <param name="align_with_fmu" type="bool" value="false"/>
    </node>

    <param name="ctrl_cmd_stream_timeout" value="0.5"/>

    <node pkg="rnw_ros" type="pub_simulated_sensors" name="pub_simulated_sensors">
        <remap from="~gps" to="/djiros/gps" />
        <remap from="~imu_in" to="/djiros/imu" />
        <remap from="~imu_out" to="/djiros/imu_sim" />
        <remap from="~odom" to="/drone_odom" />
    </node>

    <node pkg="rnw_ros" type="thrust_test_node" name="thrust_test_node">
        <remap from="~ctrl" to="/djiros/ctrl"/>
        <remap from="~flight_status" to="/djiros/flight_status"/>
    </node>

    <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_reconfigure"/>

</launch>
